### PR TITLE
fix: getLogLevel return type

### PR DESCRIPTION
**Pull request by skjnldsv**
Fixes https://github.com/nextcloud/server/pull/43534

### CODE REVIEW

**Review by SystemKeeper — APPROVED** [review]

**Comment by susnux** [comment]
Ah yes had the same issue as something set the default to `'0'`. But maybe we should add some logging for this case to warn the user that the log level config is invalid?

**Review by susnux — APPROVED** [review]
Works but I think the logging (see commit) would make sense

File: lib/private/Log.php [hunk 1]
```diff
@@ -281,7 +281,8 @@ public function getLogLevel($context): int {
 			}
 		}
 
-		return min($this->config->getValue('loglevel', ILogger::WARN) ?? ILogger::WARN, ILogger::FATAL);
+		$configLogLevel = $this->config->getValue('loglevel', ILogger::WARN);
+		return min(is_int($configLogLevel) ? $configLogLevel : ILogger::WARN, ILogger::FATAL);
```
Comment (come-nc) [inline lib/private/Log.php:285]: As @susnux pointed out we should log an error or cast the value to int.

Comment (nickvergessen) [reply lib/private/Log.php:285]: Ref https://github.com/nextcloud/server/pull/44262

**Pull request merged**